### ISSUE TITLE
Measure test coverage

### ISSUE THREAD

**Issue by JamieMagee** [dependabot/dependabot-core#9203]
### Code improvement description

Test coverage is a simple measure of how much of the codebase has been tested. I think that setting specific coverage levels is not very helpful beyond a certain point, but it can help identify parts of the codebase that are poorly tested and may require more attention.

The most popular library for measuring code coverage in Ruby is [simplecov][1].

[1]: https://github.com/simplecov-ruby/simplecov

**Comment by yeikel** [comment]
Can you assign this to me?

**Comment by JamieMagee** [comment]
@yeikel done. Thank you!

**Comment by smoookeeey** [comment]
Can you assign this to me?

**Comment by JamieMagee** [comment]
@smoookeeey I assigned you too, but @yeikel asked to be assigned first. Please make sure you don't duplicate any work @yeikel might have already done.

**Comment by smoookeeey** [comment]
Yes I will make sure not too do that

**Comment by yeikel** [comment]
I haven't started yet, and I won't have capacity until sometime next week. @smoookeeey, if you pick it up before that, feel free to create the PR and link it here

**Comment by smoookeeey** [comment]
I will look into it brother

**Comment by abdulapopoola** [comment]
Thanks a lot @yeikel and @smoookeeey !

**Comment by JamieMagee** [comment]
I just discovered that test coverage was previously added in #2576 to prevent a regression. Originally line coverage was set to 80% and branch coverage to 70%, with `refuse_coverage_drop` enabled to prevent any drops from occuring.

It was later gated behind the `COVERAGE` environment variable in #3339. However, because that environment variable wasn't set in GitHub Actions, code coverage was allowed to deteriorate. Subsequently, test coverage was removed in #7473.

**Comment by yeikel** [comment]
I think that rather than enforcing certain %, we can start reporting it for awareness

**Comment by JamieMagee** [comment]
I agree. Reporting-only is a good start. Then something like [`refuse_coverage_drop`][1] to prevent backslides and act like [a ratchet][2]

[1]: https://www.rubydoc.info/gems/simplecov/SimpleCov%2FConfiguration:refuse_coverage_drop
[2]: https://qntm.org/ratchet

**Comment by yeikel** [comment]
@smoookeeey Are you still working on this? Feel free to create a draft PR whenever you're ready

**Comment by JamieMagee** [comment]
@yeikel I think you can go ahead with this if you want.
